reflection.md

```diff
@@ -1,18 +1,45 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+The Char implicit definition was the easiest to implement. This was because
+it were basically already defined in the language as a Literal, it's just 
+slightly clunkier to write out `Literal ('c')` rather than just `'c'` so
+implicit definitions made it much easier.
```

Comment:
True! Although I struggled a little more because I had not idea how to start, once that was figured out it probably can't get easier than one function call